### DISCUSSION TITLE
How to install GTK on macOS for building Transmission? (No package 'gtkmm-4.0' found)

### DISCUSSION THREAD

**Discussion by Coeur** [transmission/transmission#3981] [Support] [answered]
I wish to build the GTK Transmission app on macOS.
I would use `cmake -B build -G Ninja -DENABLE_GTK=ON -DENABLE_MAC=OFF` (and `ninja -C build transmission-gtk` afterward).

But beforehand, I need GTK on macOS.
I tried two installation types:
A. The [official one](https://wiki.gnome.org/Projects/GTK/OSX/Building) (with `sh ./gtk-osx-setup.sh`)
B. The brew one (with `brew install gtk+3` or `brew install gtk4`)
I verified that GTK was installed with [this StackOverflow answer](https://stackoverflow.com/a/38919482):
```
$ pkg-config gtk+-3.0 --exists && echo Found the library
Found the library
```

I also tried independently:
```
export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig
export PKG_CONFIG_PATH=/opt/homebrew/lib/pkgconfig
```

But I always end up with:
```
$ cmake -B build -G Ninja -DENABLE_GTK=ON -DENABLE_MAC=OFF
-- Checking for modules 'gtkmm-4.0>=3.24.0;glibmm-2.68>=2.60.0;giomm-2.68>=2.26.0'
--   No package 'gtkmm-4.0' found
--   No package 'glibmm-2.68' found
--   No package 'giomm-2.68' found
-- Checking for modules 'gtkmm-3.0>=3.24.0;glibmm-2.4>=2.60.0;giomm-2.4>=2.26.0'
--   No package 'gtkmm-3.0' found
--   No package 'glibmm-2.4' found
--   No package 'giomm-2.4' found
CMake Error at CMakeLists.txt:289 (message):
  GTK is required but wasn't found
```

**Comment by GaryElshaw** [comment, marked as answer]
I'm using homebrew, and there are a lot of dependencies involved. I'd try:

brew install gtk+3 gtk4 gtkmm gtkmm3 gtkmm4 cairomm

It should resolve all of the dependencies.

**Reply by Coeur** [reply]
Nice. Success with `brew install gtk+3 gtkmm3`. I'll make a doc now.

**Comment by mikedld** [comment]
Not that I'm against it, but is there a use-case where you'd prefer GTK client over Mac client on macOS? :) I guess it's just to be able to make sure it builds locally before pushing the changes and opening a PR, but you'll soon see there're some issues with it that need fixing (like e.g. all items in main menu being disabled).

**Reply by Coeur** [reply]
It's just for developers, especially when making changes to core library, we occasionally need to make sure that it doesn't break the various clients (macOS, GTK, Qt, ...).

And also to screen issues, to analyse and fix bugs from all apps.

**Reply by GaryElshaw** [reply]
Absolutely. But it's great to be able to check things out. I don't know if @nevack or @ckerr will remember, but i recently discovered that all of the work i was putting into the GTK tray icon was wasted because it entirely depended on what theme the user had enabled. We were all seeing different things and pointing fingers.

**Reply by barracuda156** [reply]
@mikedld I have bumped into this topic looking for a fix for newer Intel builds, but nevertheless:

> Not that I'm against it, but is there a use-case where you'd prefer GTK client over Mac client on macOS? :)

_Yes, on macOS PowerPC!_
